pkg/security/probe/mount_resolver.go

```diff
@@ -380,17 +383,46 @@ func (mr *MountResolver) Start(ctx context.Context) {
 	}()
 }
 
-// GetMountPath returns the path of a mount identified by its mount ID. The first path is the container mount path if
-// it exists, the second parameter is the mount point path, and the third parameter is the root path.
-func (mr *MountResolver) GetMountPath(mountID uint32) (string, string, string, error) {
+func (mr *MountResolver) resolveMount(mountID, pid uint32) (*model.MountEvent, error) {
 	if mountID == 0 {
-		return "", "", "", nil
+		return nil, ErrMountUndefined
 	}
-	mr.lock.RLock()
-	defer mr.lock.RUnlock()
 
 	mount, ok := mr.mounts[mountID]
+
 	if !ok {
+		mr.cacheMissStats.Inc()
+		if pid != 0 {
+			if err := mr.syncCache(pid); err != nil {
+				return nil, err
+			}
+			mount = mr.mounts[mountID]
+			if mount != nil {
+				mr.procHitsStats.Inc()
+			} else {
+				mr.procMissStats.Inc()
+			}
+		}
+	} else {
+		// stats
+		mr.cacheHitsStats.Inc()
+	}
+
+	if mount == nil {
+		return nil, ErrMountNotFound
```

Comment:
should we add a counter to track this case ? I think it could help to see how much error we have